spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverter.java

```diff
@@ -79,21 +85,35 @@ public DefaultDatastoreEntityConverter(DatastoreMappingContext mappingContext,
 	}
 
 	@Override
+	@SuppressWarnings("unchecked")
 	public void write(Object source, Entity.Builder sink) {
 		DatastorePersistentEntity<?> persistentEntity = this.mappingContext.getPersistentEntity(source.getClass());
 		PersistentPropertyAccessor accessor = persistentEntity.getPropertyAccessor(source);
 		persistentEntity.doWithProperties(
 				(DatastorePersistentProperty datastorePersistentProperty) ->
-						writeProperty(sink, accessor, datastorePersistentProperty));
+						writeProperty(
+								sink, accessor.getProperty(datastorePersistentProperty), datastorePersistentProperty));
 	}
 
-	private void writeProperty(Entity.Builder sink, PersistentPropertyAccessor accessor,
+	private void writeProperty(Entity.Builder sink, Object propertyVal,
 			DatastorePersistentProperty persistentProperty) {
-		Object propertyVal = accessor.getProperty(persistentProperty);
+		boolean isArray = propertyVal != null && Object[].class.isAssignableFrom(propertyVal.getClass());
```

Comment:
Maybe add a comment or turn the two lines into a utility method - `wrapArrayIntoList`.